src/local_sensitivity/backsolve_adjoint.jl

```diff
@@ -82,6 +97,70 @@ end
   return ODEProblem(odefun,z0,tspan,p,callback=cb)
 end
 
+
+
+@noinline function SDEAdjointProblem(sol,sensealg::BacksolveAdjoint,
+                                     g,t=nothing,dg=nothing;
+                                     checkpoints=sol.t,
+                                     callback=CallbackSet(),kwargs...)
+  @unpack f, p, u0, tspan = sol.prob
+  tspan = reverse(tspan)
+  discrete = t != nothing
+
+  p === DiffEqBase.NullParameters() && error("Your model does not have parameters, and thus it is impossible to calculate the derivative of the solution with respect to the parameters. Your model must have parameters to use parameter sensitivity calculations!")
+
+  !(StochasticDiffEq.is_diagonal_noise(sol.prob))  && error("Your model has non-diagonal noise terms. Only scalar and diagonal noise terms are currently supported.")
+
+  numstates = length(u0)
+  numparams = length(p)
+
+  len = length(u0)+numparams
+  λ = similar(p, len)
+
+  sense_drift = ODEBacksolveSensitivityFunction(g,sensealg,discrete,sol,dg,sol.prob.f,sol.prob.f.colorvec)
+
+  diffusion_function = ODEFunction(sol.prob.g)
+  sense_diffusion = ODEBacksolveSensitivityFunction(g,sensealg,discrete,sol,dg,diffusion_function,diffusion_function.colorvec)
+
+  init_cb = t !== nothing && tspan[1] == t[end]
+  cb = generate_callbacks(sense_drift, g, λ, t, callback, init_cb)
+  checkpoints = ischeckpointing(sensealg, sol) ? checkpoints : nothing
+  if checkpoints !== nothing
+    cb = backsolve_checkpoint_callbacks(sense_drift, sol, checkpoints, cb)
+  end
+
+  z0 = [vec(zero(λ)); vec(sense_drift.y)]
+
+  original_mm = sol.prob.f.mass_matrix
+  if original_mm === I
+    mm = I
+  else
+    sense_drift.diffcache.issemiexplicitdae && @warn "`BacksolveAdjoint` is likely to fail on semi-explicit DAEs, if memory is a concern, please consider using InterpolatingAdjoint(checkpoint=true) instead."
+    len2 = length(z0)
+    mm = zeros(len2, len2)
+    idx = 1:numstates
+    copyto!(@view(mm[idx, idx]), sol.prob.f.mass_matrix')
+    idx = numstates+1:numstates+1+numparams
+    copyto!(@view(mm[idx, idx]), I)
+    idx = len+1:len2
+    copyto!(@view(mm[idx, idx]), sol.prob.f.mass_matrix)
+  end
+
+  sdefun = SDEFunction(sense_drift,sense_diffusion,mass_matrix=mm)
+
+  # replicated noise
+  _sol = deepcopy(sol)
+  backwardnoise = DiffEqNoiseProcess.NoiseGrid(reverse!(_sol.t),reverse!( _sol.W.W))
```

Comment:
we probably need to add `reverse!(::AbstractNoiseProcess)` to generalize this. Open an issue on DiffEqNoiseProcess to track that.